app/pages/lab/workflow-viewer/workflow.cjsx

```diff
@@ -0,0 +1,226 @@
+        when 'multiple'
+          if task.subtask
+            if task.next?
+              c = ["#{id}_next", task.next]
+              @props.jp.connect({uuids: c, detachable: DETACHABLE})
+              @appendToDict(next_task, id, c[1])
+              @appendToDict(previous_task, c[1], id)
+          else
+            c = ["#{id}_next", task.next ? 'end']
+            @props.jp.connect({uuids: c, detachable: DETACHABLE})
+            @appendToDict(next_task, id, c[1])
+            @appendToDict(previous_task, c[1], id)
+        when 'drawing'
+          c = ["#{id}_next", task.next ? 'end']
+          @props.jp.connect({uuids: c, detachable: DETACHABLE})
+          @appendToDict(next_task, id, c[1])
+          @appendToDict(previous_task, c[1], id)
+          for a, adx in task.tools
+            if a.details[0]?
+              c = ["#{id}_answer_#{adx}", a.details[0]]
+              @props.jp.connect({uuids: c, detachable: DETACHABLE})
+              @appendToDict(next_task, id, c[1])
+              @appendToDict(previous_task, c[1], id)
+      next_task[id] ?= {'end': 'end'}
+    @setState({next_task: next_task, previous_task: previous_task})
+    #@sortTasks(connections)
+
+  getUuid: (idx, uuid = @state.uuid, uuids = @state.uuids) ->
+    if uuids[idx]?
+      return uuids[idx]
+    else
+      return "T#{uuid}"
+
+  setUuid: (id) ->
+    current_uuids = @state.uuids.concat([id])
+    current_uuid = @state.uuid + 1
+    @setState({uuids: current_uuids, uuid: current_uuid})
+
+  getWorkflow: ->
+    # format workflow so 'init' is not a key and sub-tasks are their own tasks
+    keys = []
+    taskStateSet = {}
+    wf = {}
+    init = undefined
+    #L = Object.keys(@props.workflow.tasks).length
+    L = 0
+    for k, v of @props.workflow.tasks
+      if k == 'init'
+        # find a new name for 'init' (lowest task number not in use)
+        ct = 0
+        tmp = 'T' + ct
+        while tmp of @props.workflow.tasks
+          ct += 1
+          tmp = 'T' + ct
+        k = tmp
+        init = k
+      keys.push(k)
+      taskStateSet[k] = false
+      # clone the workflow so it does not overwrite the original (will update the API after re-formatting)
+      wf[k] = clone(v)
+      wf[k].subTask = false
+      if v.type == 'drawing'
+        for tool, tdx in v.tools
+          subList = []
+          # make subtasks into their own tasks (use lowest task number not in use)
+          for st, sdx in tool.details
+            tmp = 'S' + L
+            subList.push(tmp)
+            keys.push(tmp)
+            wf[tmp] = clone(st)
+            wf[tmp].subtask = true
+            L += 1
+            # set 'next' to be the next subtask in the list (if it exists)
+            if tool.details[sdx + 1]?
+              wf[tmp].next = 'S' + L
+          # replace details with a list of keys
+          wf[k].tools[tdx].details = subList
+    # make sure the keys are in number order
+    keys.sort()
+    {init, keys, wf, taskStateSet}
```

Comment:
The reason I am doing this here is because I need to extract the sub-tasks as full nodes.  For the visualization bit this is not that important, but when I extend this into a full editor it will become important (i.e., when the user disconnects a sub-task the node should stick around as a normal task in the workflow, or a normal task can be connected up as a sub-task).